source/Font.h

```diff
@@ -36,14 +36,29 @@ class Font {
 	
 	void Load(const std::string &imagePath);
 	
-	void Draw(const std::string &str, const Point &point, const Color &color) const;
-	void DrawAliased(const std::string &str, double x, double y, const Color &color) const;
+	// Layout parameters.
+	enum Align {LEFT, CENTER, RIGHT};
+	enum Truncate {TRUNC_NONE, TRUNC_FRONT, TRUNC_MIDDLE, TRUNC_BACK};
+	struct Layout {
+			// Align and trancate width. No align or trancate if width is negative.
+			int width = -1;
+			// Set the alignment mode.
+			Align align = LEFT;
+			// Set the truncate mode.
+			Truncate truncate = TRUNC_NONE;
+			
+			Layout() noexcept;
+			Layout(int w, Align a) noexcept;
+			Layout(int w, Truncate t) noexcept;
+			Layout(int w, Truncate t, Align a) noexcept;
```

Comment:
question: Should this be width - Align - Truncate? Or is width - Truncate - Align more "natural," in call-site usage?

Reply:
I feel Align is more important, so width - Align - Truncate.